filebeat/crawler/registrar.go

```diff
@@ -63,14 +64,84 @@ func (r *Registrar) Init() error {
 
 // loadState fetches the previous reading state from the configure RegistryFile file
 // The default file is `registry` in the data path.
-func (r *Registrar) LoadState() {
-	if existing, e := os.Open(r.registryFile); e == nil {
-		defer existing.Close()
+func (r *Registrar) LoadState() error {
+
+	// Check if files exists
+	_, err := os.Stat(r.registryFile)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
+	// Error means no file found
+	if err != nil {
```

Comment:
It could also happen, that filebeat is restarted but does not have file write permission anymore, means registry file aready exist but filebeat was started last time by a different user. Should we also exit? I suggest yes.

I have to find an easy way to determine if we can write to the file / directory :-)